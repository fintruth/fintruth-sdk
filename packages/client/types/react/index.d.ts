import 'react'
import { CSSProp, DefaultTheme } from 'styled-components' // eslint-disable-line import/named

declare module 'react' {
  interface Attributes {
    css?: CSSProp<DefaultTheme>
  }
}
