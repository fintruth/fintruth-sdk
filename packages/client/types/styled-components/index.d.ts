import 'styled-components'

declare module 'styled-components' {
  export interface Body {
    color: string
    fontFamily: string
    fontWeight: number
    lineHeight: number
  }

  export interface Code {
    backgroundColor: string
    color: string
    fontFamily: string
    fontSize: string
    fontWeight: number
    padding: string
  }

  export interface DefaultTheme extends Variables {
    body: Body
    code: Code
    hr: Hr
    html: Html
    pre: Pre
    strong: Strong
  }

  export interface Hr {
    backgroundColor: string
    height: string
    margin: string
  }

  export interface Html {
    backgroundColor: string
    fontSize: string
    overflowX: string
    overflowY: string
    textRendering: string
  }

  export interface Pre {
    backgroundColor: string
    color: string
  }

  export interface Strong {
    color: string
    fontWeight: number
  }

  export interface Variables {
    backgroundColor: string
    black: string
    blackBis: string
    blackTer: string
    blue: string
    blueContrast: string
    borderColor: string
    borderHoverColor: string
    cyan: string
    cyanContrast: string
    danger: string
    dangerContrast: string
    darkContrast: string
    fontFamilyCode: string
    fontFamilyMonospace: string
    fontFamilyPrimary: string
    fontFamilySansSerif: string
    fontFamilySecondary: string
    gap: number
    gray: string
    grayDark: string
    grayDarker: string
    grayLight: string
    grayLighter: string
    green: string
    greenContrast: string
    info: string
    infoContrast: string
    lightContrast: string
    linkActiveBorderColor: string
    linkActiveColor: string
    linkColor: string
    linkColorContrast: string
    linkFocusBorderColor: string
    linkFocusColor: string
    linkHoverBorderColor: string
    linkHoverColor: string
    linkVisitedColor: string
    orange: string
    orangeContrast: string
    primary: string
    primaryContrast: string
    purple: string
    purpleContrast: string
    red: string
    redContrast: string
    success: string
    successContrast: string
    textColor: string
    textColorContrast: string
    textLightColor: string
    textRendering: string
    textSelectionColor: string
    textStrongColor: string
    turquoise: string
    turquoiseContrast: string
    viewportExtraLarge: number
    viewportLarge: number
    viewportMedium: number
    viewportSmall: number
    warning: string
    warningContrast: string
    white: string
    whiteBis: string
    whiteTer: string
    yellow: string
    yellowContrast: string
  }
}
