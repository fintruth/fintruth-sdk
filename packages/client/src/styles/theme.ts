import { getLuminance, hsl, hsla, rem } from 'polished'
import { Color, ColorContrast, DefaultTheme, Variant } from 'styled-components'

const contrastColor = (color: string) =>
  getLuminance(color) > 0.55 ? hsla(0, 0, 0, 0.7) : hsl(0, 0, 1)

const grayDarker = hsl(0, 0, 0.21)
const grayDark = hsl(0, 0, 0.29)
const gray = hsl(0, 0, 0.48)
const grayLight = hsl(0, 0, 0.71)
const grayLighter = hsl(0, 0, 0.86)
const grayLightest = hsl(0, 0, 0.93)

const whiteTer = hsl(0, 0, 0.96)

const light = whiteTer
const dark = grayDarker

const orange = hsl(14, 1, 0.53)
const yellow = hsl(48, 1, 0.67)
const green = hsl(141, 0.71, 0.48)
const turquoise = hsl(171, 1, 0.41)
const cyan = hsl(204, 0.86, 0.53)
const blue = hsl(217, 0.71, 0.53)
const purple = hsl(271, 1, 0.71)
const red = hsl(348, 1, 0.61)

const blueContrast = contrastColor(blue)

const primary = turquoise
const info = cyan
const success = green
const warning = yellow
const danger = red

const backgroundColor = whiteTer

const textColor = grayDark

const fontFamilySansSerif =
  'BlinkMacSystemFont, -apple-system, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", "Helvetica", "Arial", sans-serif'
const fontFamilyMonospace = 'monospace'

const gap = 32

const theme: DefaultTheme = {
  black: hsl(0, 0, 0.04),
  blackBis: hsl(0, 0, 0.07),
  blackTer: hsl(0, 0, 0.14),

  grayDarker,
  grayDark,
  gray,
  grayLight,
  grayLighter,
  grayLightest,

  whiteTer,
  whiteBis: hsl(0, 0, 0.98),
  white: hsl(0, 0, 1),

  light,
  dark,

  lightContrast: dark,
  darkContrast: light,

  orange,
  yellow,
  green,
  turquoise,
  cyan,
  blue,
  purple,
  red,

  orangeContrast: contrastColor(orange),
  yellowContrast: contrastColor(yellow),
  greenContrast: contrastColor(green),
  turquoiseContrast: contrastColor(turquoise),
  cyanContrast: contrastColor(cyan),
  blueContrast,
  purpleContrast: contrastColor(purple),
  redContrast: contrastColor(red),

  primary,
  info,
  success,
  warning,
  danger,

  primaryContrast: contrastColor(primary),
  infoContrast: contrastColor(info),
  successContrast: contrastColor(success),
  warningContrast: contrastColor(warning),
  dangerContrast: contrastColor(danger),

  backgroundColor,

  borderColor: grayLighter,
  borderHoverColor: grayLight,
  borderLightColor: grayLightest,
  borderLightHoverColor: grayLight,

  textColor,
  textColorContrast: contrastColor(textColor),
  textLightColor: gray,
  textStrongColor: grayDarker,
  textSelectionBackgroundColor: hsl(213, 0.92, 0.85),

  codeColor: red,
  codeBackgroundColor: backgroundColor,

  preColor: textColor,
  preBackgroundColor: backgroundColor,

  linkColor: blue,
  linkColorContrast: blueContrast,
  linkVisitedColor: purple,

  linkHoverColor: grayDarker,
  linkHoverBorderColor: grayLight,

  linkFocusColor: grayDarker,
  linkFocusBorderColor: blue,

  linkActiveColor: grayDarker,
  linkActiveBorderColor: grayDark,

  fontFamilySansSerif,
  fontFamilyMonospace,
  textRendering: 'optimizeLegibility',

  fontFamilyPrimary: fontFamilySansSerif,
  fontFamilySecondary: fontFamilySansSerif,
  fontFamilyCode: fontFamilyMonospace,

  blockSpacing: rem(24),

  gap,
  small: 769,
  medium: 960 + 2 * gap,
  large: 1152 + 2 * gap,
  extraLarge: 1344 + 2 * gap,

  timingFunction: 'ease-out',
  borderRadiusSmall: '2px',
  borderRadius: '4px',
  borderRadiusLarge: '6px',
  borderRadiusRounded: '999999px',
  duration: '86ms',
}

export const variantColors: Record<Variant, Color> = {
  danger: 'danger',
  dark: 'dark',
  info: 'info',
  light: 'light',
  primary: 'primary',
  success: 'success',
  warning: 'warning',
}

export const variantColorContrasts: Record<Variant, ColorContrast> = {
  danger: 'dangerContrast',
  dark: 'darkContrast',
  info: 'infoContrast',
  light: 'lightContrast',
  primary: 'primaryContrast',
  success: 'successContrast',
  warning: 'warningContrast',
}

export default theme
