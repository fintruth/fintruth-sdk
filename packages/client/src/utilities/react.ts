import React from 'react'

export const setRef = <T>(
  ref: React.Ref<T> | React.MutableRefObject<T | undefined>,
  instance: T | null
) => {
  if (typeof ref === 'function') {
    ref(instance)
  } else if (ref !== null && instance !== null) {
    // @ts-ignore readonly-property
    ref.current = instance // eslint-disable-line no-param-reassign
  }
}
