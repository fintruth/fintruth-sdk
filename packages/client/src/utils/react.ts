import React from 'react'

export const setRef = <T>(
  ref: React.Ref<T> | React.MutableRefObject<T | undefined>,
  instance: T | null
) => {
  if (typeof ref === 'function') {
    ref(instance)
  } else if (ref !== null && instance !== null) {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore TS2540
    ref.current = instance // eslint-disable-line no-param-reassign
  }
}
