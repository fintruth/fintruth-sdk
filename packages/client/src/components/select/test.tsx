import { render } from '@testing-library/react'
import React from 'react'
import { ThemeProvider } from 'styled-components'

import theme from 'styles/theme'
import Select from '.'

test('should render correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render correctly when disabled', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select isDisabled />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render correctly when disabled and multiple', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select isDisabled isMultiple />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select variant="danger" />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly when disabled', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select isDisabled variant="danger" />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly when multiple', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Select isMultiple variant="danger" />
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})
