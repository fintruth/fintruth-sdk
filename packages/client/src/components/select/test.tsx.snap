// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c0::after {
  border: 5px solid transparent;
  border-bottom-width: 0;
  border-top-color: #3273dc;
  content: '';
  display: block;
  pointer-events: none;
  position: relative;
  position: absolute !important;
  right: 0.8125em;
  top: calc(50% - 2.5px);
  z-index: 4;
}

.c0:hover::after {
  border-top-color: #363636;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #dbdbdb;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #b5b5b5;
}

.c1:focus,
.c1:active {
  border-color: #3273dc;
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
  />
</div>
`;

exports[`should render correctly when disabled 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #dbdbdb;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #b5b5b5;
}

.c1:focus,
.c1:active {
  border-color: #3273dc;
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
    disabled=""
  />
</div>
`;

exports[`should render correctly when disabled and multiple 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #dbdbdb;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #b5b5b5;
}

.c1:focus,
.c1:active {
  border-color: #3273dc;
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
    disabled=""
    multiple=""
  />
</div>
`;

exports[`should render the danger variant correctly 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c0::after {
  border: 5px solid transparent;
  border-bottom-width: 0;
  border-top-color: #ff3860;
  content: '';
  display: block;
  pointer-events: none;
  position: relative;
  position: absolute !important;
  right: 0.8125em;
  top: calc(50% - 2.5px);
  z-index: 4;
}

.c0:hover::after {
  border-top-color: #363636;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #ff3860;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #ff2b56;
}

.c1:focus,
.c1:active {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
  />
</div>
`;

exports[`should render the danger variant correctly when disabled 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #ff3860;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #ff2b56;
}

.c1:focus,
.c1:active {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
    disabled=""
  />
</div>
`;

exports[`should render the danger variant correctly when multiple 1`] = `
.c0 {
  max-width: 100%;
  position: relative;
  vertical-align: top;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px rgba(10,10,10,0.1);
  color: #363636;
  cursor: pointer;
  display: block;
  font-size: 1em;
  max-width: 100%;
  outline: none;
  padding-right: 2.25em;
  width: 100%;
  border-color: #ff3860;
}

.c1:focus,
.c1:active {
  outline: none;
}

.c1[disabled],
fieldset[disabled] .c1 {
  cursor: not-allowed;
}

.c1:hover {
  border-color: #ff2b56;
}

.c1:focus,
.c1:active {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c1:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #363636;
}

.c1::-ms-expand {
  display: none;
}

.c1[multiple] {
  height: auto;
  padding: 0;
}

.c1[disabled],
fieldset[disabled] .c1 {
  background-color: #f5f5f5;
  border-color: #f5f5f5;
  box-shadow: none;
  color: #7a7a7a;
}

<div
  class="c0"
>
  <select
    autocomplete="off"
    class="c1"
    multiple=""
  />
</div>
`;
